packages/components/src/lib/MarkdownViewer.tsx

```diff
@@ -0,0 +1,76 @@
+import { Element } from "hast";
+import React from "react";
+import ReactMarkdown from "react-markdown";
+import { Prism as SyntaxHighlighter } from "react-syntax-highlighter";
+import * as style from "react-syntax-highlighter/dist/esm/styles/prism";
+import remarkGfm from "remark-gfm";
+import { Node, Parent } from "unist";
+import { visitParents } from "unist-util-visit-parents";
+
+// Adds `inline` property to `code` elements, to distinguish between inline and block code snippets.
+function rehypeInlineCodeProperty() {
```

Comment:
This seems kind of gnarly, but was the main solution I could find online at this time. I also used ChatGPT to help convert the JS solution to TS.

Reply:
Types seem messy to me (too many explicit `as` conversions), but if it works then I don't mind.